packages/client/hmi-client/src/views/ModelView.vue

```diff
@@ -1,28 +1,37 @@
 <script setup lang="ts">
-import Model from '@/components/models/Model.vue';
+import Model, { ModelProps } from '@/components/models/Model.vue';
 import TabContainer from '@/components/tabs/TabContainer.vue';
-import { ref, watch, computed, onMounted } from 'vue';
+import { ref, watch, computed } from 'vue';
 import { Tab } from '@/types/common';
 import useResourcesStore from '@/stores/resources';
-
-interface ModelProps {
-	modelId: string;
-}
+import { useTabStore } from '@/stores/tabs';
 
 const props = defineProps<{
-	modelId: string;
+	modelId?: string;
 }>();
 
 const resourcesStore = useResourcesStore();
+const tabStore = useTabStore();
 
 const newModelId = computed(() => props.modelId);
 const openTabs = ref<Tab[]>([]);
 const activeTabIndex = ref(0);
 const modelsInCurrentProject = resourcesStore.activeProjectAssets?.models;
+const activeProject = resourcesStore.activeProject;
+const tabContext = `model${activeProject?.id}`;
+
+// @ts-ignore
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+tabStore.$subscribe((mutation, state) => {
+	const savedTabs = state.tabMap.get(tabContext);
+	if (savedTabs) {
+		openTabs.value = savedTabs;
+	}
+	activeTabIndex.value = tabStore.getActiveTabIndex(tabContext);
+});
```

Comment:
Could you explain what this `$subscribe` method and what does it do.
Also, let's try to not use `@ts-ignore`

Reply:
Unfortunately, I have to here because of the `mutation` variable, which I have to include (just declaring `state` does not work)